### ISSUE TITLE
package the GUI as a separate binary

### ISSUE THREAD

**Issue by flokli** [sahib/rmlint#419]
Motivated by https://github.com/NixOS/nixpkgs/pull/90267.

The way `rmlint` currently starts its GUI is somewhat weird.

It writes a `.shredder-bootstrap.py.XXXXXX` file in `/tmp` containing some python code manually appending things to `sys.path`, then running

```
import shredder
shredder.run_gui()
```

This causes hard-to-debug problems for platforms where shredder isn't installed into `PYTHONPATH`, like https://github.com/sahib/rmlint/issues/253.

Also, it makes it quite complicated for packagers to move the GUI bits into a separate package output.

Instead, what about shipping `shredder` as a separate binary, with entrypoints managed via `setuptools`?

`rmlint --gui` could still be supported, by executing `shredder` from `$PATH`.

**Comment by CouldBeThis** [comment]
I installed via debian `apt-get` and noticed there is one package for [`rmlint`](https://packages.debian.org/buster/rmlint), one for [`rmlint-gui`](https://packages.debian.org/buster/rmlint-gui) and another for [`rmlint-doc`](https://packages.debian.org/buster/rmlint-doc). 

A menu item is added to the GUI: `Applications > System Tools > Shredder`

Just mentioning cause this issue is still open.

**Comment by SeeSpotRun** [comment]
Hey thanks @CouldBeThis and @flokli 

We dropped the `rmlint` ball a bit last year and @sahib asked me to take over maintenance at the start of this year.  I'm working my way back through open issues now and hadn't gotten back to this one.

Python / `setuptools` is not my strong suit so would be happy to see a pull request on this one.

**Comment by SeeSpotRun** [comment]
This issue is now partially resolved; there is no longer a python bootstrap in /tmp, and there's a [slightly-hacky workaround](https://github.com/sahib/rmlint/blob/1a142fc0c4fd5239b99cb541863970f0c6f6bfee/gui/bin/shredder#L7-L12) in /bin/shredder to reverse-engineer `PYTHONPATH`.

I'm not planning anything further at this stage I think there's still a case for fully splitting the package so that `rmlint` installs (without gui) via
```
$ scons install rmlint
```

and `shredder` installs via
```
$ python setup.py install
```
(or possibly via `pip install`)

**Comment by AutomationD** [comment]
Thanks for tremendous work! Is there any tested alpine setup? So far I got to the shredder, but can't find GtkSource. 

Here's the list of package I'm installing:
```
apk add	scons \
	glib \
	glib-dev \
	json-glib \
	json-glib-dev \
	libelf \
	libstdc++ \
	py3-sphinx \
	elfutils-dev \
	py3-gobject3 \
	gobject-introspection \
	gobject-introspection-dev \
	py3-gobject3-dev \
	gtk+3.0-dev \
	gdk-pixbuf \
	gdk-pixbuf-dev
```

**Comment by SeeSpotRun** [comment]
> any tested alpine setup? 

...have just set up a VM now.

I had to add to your list:
```
        build-base \
        librsvg \
        gtksourceview \
        py3-cairo
```

**Comment by AutomationD** [comment]
Hey, @SeeSpotRun , Thanks,  I got it working too! 
So far one of the best solutions for remote-only datahoarders who want a GUI, running via X...

For anyone wondering, the full list of deps for alpine would be:
```
apk add	scons \
	glib \
	glib-dev \
	json-glib \
	json-glib-dev \
	libelf \
	libstdc++ \
	py3-sphinx \
	elfutils-dev \
	py3-gobject3 \
	gobject-introspection \
	gobject-introspection-dev \
	py3-gobject3-dev \
	gtk+3.0-dev \
	gdk-pixbuf \
	gdk-pixbuf-dev
        build-base \
        librsvg \
        gtksourceview \
        py3-cairo
```

**Comment by vassilit** [comment]
Fixed, now we ship usr/bin/shredder

**Comment by RayOei** [comment]
True, but I was more interpreting this [comment](https://github.com/sahib/rmlint/issues/419#issuecomment-849964421) too.
If we consider only the binary itself I suppose this issue can be closed?

The install will also be addressed by #732 I assume? As `distutils` has also been deprecated it seems the GUI install needs more attention anyway.

**Comment by vassilit** [comment]
> True, but I was more interpreting this [comment](https://github.com/sahib/rmlint/issues/419#issuecomment-849964421) too. If we consider only the binary itself I suppose this issue can be closed?

`scons install --without-gui` and `cd gui && python setup.py install` works.

However, `pip install` does not work because of the deprecated distutils dependency for install_data that have no direct replacement with setuptools.
